src/GraphMetadataViewExtension/GraphMetadataViewModel.cs

```diff
@@ -107,5 +112,36 @@ private static string Base64FromImage(BitmapImage source)
 
             return Convert.ToBase64String(data);
         }
+
+        private void InitializeCommands()
+        {
+            this.AddCustomPropertyCommand = new DelegateCommand(AddCustomPropertyExecute);
+        }
+
+        private void AddCustomPropertyExecute(object obj)
+        {
+            var control = new CustomPropertyControl();
+            control.PropertyName = $"Custom Property {CustomProperties.Count + 1}";
+            control.RequestDelete += HandleDeleteRequest;
+            CustomProperties.Add(control);
+        }
+
+
+
+        private void HandleDeleteRequest(object sender, EventArgs e)
+        {
+            CustomPropertyControl customProperty = sender as CustomPropertyControl;
+
+            if(customProperty!=null)
+            {
+                customProperty.RequestDelete -= HandleDeleteRequest;
+                CustomProperties.Remove(customProperty);
+            }
+        }
+
+       public void Dispose()
```

Comment:
here we would also need to unsubscribe any CustomProperties in the `CustomProperties` collection:

```csharp
foreach(var cp in CustomProperties)
{
    cp.RequestDelete -= HandleDeleteRequest;
}